oauthenticator/oauth2.py

```diff
@@ -261,6 +267,16 @@ def _userdata_url_default(self):
         """,
     )
 
+    @default("extra_authorize_params")
+    def _extra_authorize_params(self):
+        return os.environ.get("OAUTH2_EXTRA_AUTHORIZE_PARAMS", {})
+
+    extra_authorize_params = Dict(
+        config=True,
+        help="""Extra GET params to send along with the initial OAuth request
+        to the OAuth provider.""",
+    )
+
```

Comment:
This extends their core oauth2 module for refresh tokens and sessions auth params. This should probably go in a different PR.